### DISCUSSION TITLE
Use URLs from STAC items

### DISCUSSION THREAD

**Discussion by drnextgis** [developmentseed/cogeo-mosaic#206] [Q&A]
I have a huge list of STAC items I want to build a mosaicJSON from. Currently only `feature["properties"]` can be specified with `--property` argument and there is no way to extract URLs out of the assets. Am I missing something? I could extract URLs from STAC items and use `cogeo-mosaic create` instead of `cogeo-mosaic create-from-features` but it takes hours to query footprints.

**Comment by drnextgis** [comment]
A custom snippet to achieve what I need:

```python
import json
import pathlib
from typing import Dict

from cogeo_mosaic.backends import MosaicBackend
from cogeo_mosaic.mosaic import MosaicJSON


def dem_stac_accessor(feature: Dict):
    return feature["assets"]["elevation"]["href"]


items = pathlib.Path("items")
features = []

for item in items.iterdir():
    with open(item) as src:
        features.append(json.load(src))

mosaicjson = MosaicJSON.from_features(
    features, minzoom=10, maxzoom=19, accessor=dem_stac_accessor
)

with MosaicBackend("mossaic.json", mosaic_def=mosaicjson) as mosaic:
    mosaic.write(overwrite=True)
```

@vincentsarago do you think this use-case is general enough to add its support via CLI?

**Reply by vincentsarago** [reply]
👋 thanks for starting this discussion @drnextgis 

I'm not sure how you would specify the accessor using CLI 🤷 

Right now we can pass a `property` https://github.com/developmentseed/cogeo-mosaic/blob/3319e0781015486ed81658c07df67f9b346a1f62/cogeo_mosaic/scripts/cli.py#L168 

so I think the easiest thing you could do is something like 

```
cat features.json | jq '.properties.path==.assets.cog.href' | cogeo-mosaic create-from-features - --min-zoom ... --accessor path 
```

**Reply by drnextgis** [reply]
Awesome! Thank you very much Vincent :pray: 

```
$ find ./items -name '*.json' -type f -exec cat {} + | jq '.properties.path=.assets.elevation.href' --compact-output | cogeo-mosaic create-from-features - --minzoom 10 --maxzoom 19 --property "path" -o mosaic.json
```
